leadership.go

```diff
@@ -109,7 +111,9 @@ func (l *Lock) RefreshLock(ctx context.Context) error {
 	l.logger(ctx).Info("Refreshing lock", "path", l.path)
 
 	attrs, err := l.bucket.Object(l.path).
-		If(storage.Conditions{MetagenerationMatch: l.latestMetadataGeneration}).
+		If(storage.Conditions{
+			GenerationMatch:     l.latestGeneration,
+			MetagenerationMatch: l.latestMetadataGeneration}).
```

Comment:
Do we need to have any conditions on metadata? 
I understand that we only care about the GenerationMatch, which should always match `0`.